lib/reporters/error-reporter.js

```diff
@@ -0,0 +1,36 @@
+const colors = require('colors');
+const isArray = require('lodash/isArray');
+const logger = require('../logger');
+const path = require('path');
+
+function errorReporter(error) {
+	const errors = (isArray(error)) ? error : [error];
+
+	logger.log('\nCaught errors:\n');
+
+	const errorStr = errors
+		.map(error => formatError(error))
+		.join('\n---\n');
+
+	logger.log(errorStr);
+}
+
+function formatError(error) {
```

Comment:
Use more helper variables and functions so `format` becomes simpler and more expressive.
Then use an extra filter method to take out empty format values, to keep format simple.

```javascript
function errorMessage(error) {
  const { cause, cwd, fileName, message, name, parser, plugin } = error;
  const hasLocation = (cwd & fileName);
  const getExactLocation = () => `${path.relative(cwd, fileName)}:${cause.line}:${cause.col}`;
  const getStackLocation = () => error.stack.split('\n    ').slice(1).join('\n\t ');

  const format = {
    name, 
    parsers,
    plugin,
    message: cause.message | message,
    where: hasLocation ? getExactLocation() : getStackLocation()
  };

  return Object.keys(format)
    .filter(key => format[key])
    .map(colors.red(`  ${key}`))
    .join('');
}
```